Framework/Kernel/inc/MantidKernel/EnumeratedString.h

```diff
@@ -59,12 +59,12 @@ class EnumeratedString {
 
   // assign the object either by the enum, or by string
   EnumeratedString &operator=(E e) {
-    if (size_t(e) < names->size() && size_t(e) >= 0) {
+    if (int(e) >= 0 && size_t(e) < names->size()) {
```

Comment:
I think just casting it to a size_t and ensuring `size_t(e) < names->size()` is sufficient.  I don't think we need to check that `int(e) >= 0`, since `size_t(-1)` will make a very large integer.

Reply:
I understand that, but I would still leave it there, since it makes the intent - not allowing negative enums - clearer for the reader.

Reply:
That makes sense, as a hint to devs if they encounter this error